swagger_spec_validator/util.py

```diff
@@ -2,36 +2,53 @@
 import logging
 import urllib2
 
-from .validator import validate_resource_listing
-from .validator import validate_api_declaration
+from swagger_spec_validator import validator12, validator20
+from swagger_spec_validator.common import SwaggerValidationError
 
 
 log = logging.getLogger(__name__)
 
+version_to_validator = {
+    '1.2': validator12,
+    '2.0': validator20,
+}
 
-def validate_resource_listing_url(url):
-    """Simple utility function to perform recursive validation of a Resource
-    Listing and all associated API Declarations.
 
-    This is trivial wrapper function around
-    :py:func:`swagger_spec_validator.validate_resource_listing` and
-    :py:func:`swagger_spec_validator.validate_api_declaration`.  You are
-    encouraged to write your own version of this if required.
-
-    :param url: the URL of the Resource Listing.
-
-    :returns: `None` in case of success, otherwise raises an exception.
-
-    :raises: :py:class:`swagger_spec_validator.SwaggerValidationError`
-    :raises: :py:class:`jsonschema.exceptions.ValidationError`
+def get_validator(spec_json, origin='unknown'):
     """
-
-    log.info('Validating %s' % url)
-    resource_listing = json.load(urllib2.urlopen(url, timeout=1))
-    validate_resource_listing(resource_listing)
-
-    for api in resource_listing['apis']:
-        path = url + api['path']
-        log.info('Validating %s' % path)
-        api_declaration = json.load(urllib2.urlopen(path, timeout=1))
-        validate_api_declaration(api_declaration)
+    :param spec_json: Dict representation of the json API spec
+    :param origin: filename or url of the spec - only use for error messages
+    :return: module responsible for validation based on Swagger version in the
+        spec
+    """
+    swagger12_version = spec_json.get('swaggerVersion')
+    swagger20_version = spec_json.get('swagger')
+
+    if swagger12_version and swagger20_version:
+        raise SwaggerValidationError(
+            "You've for conflicting keys for the Swagger version in your spec. "
+            "Expected `swaggerVersion` or `swagger`, but not both.")
+    elif swagger12_version or swagger20_version:
+        swagger_version = swagger12_version or swagger20_version
```

Comment:
I think this is wrong. `swagger20_version` could have been given a value `1.2` and would be accepted as a valid `swagger1.2`, but as per 2.0 spec `swagger` can only hold `2.0`. The two should be checked separately.